<?xml version="1.0" ?>
<launch>
	<param name="use_sim_time" value="false" />
	
	<!-- arguments -->
	<arg name="load_gui" default="true" /> <!-- Loading GUI : Enables seeing the joints position -->
	<arg name="use_gui" default="false" /> <!-- Using GUI : Enables changing from the gui the joints position -->
	<arg name="use_hocu" default="false" /><!-- Using Hocu : Enables using HDT Hocu Controller -->
	<arg name="use_leap" default="false" /><!-- Using Leap Motion : Enables using Leap Motion Controller -->
	<arg name="use_axis" default="true" /><!-- Using Axis Camera : Enables using the Axis Camera -->
	<arg name="use_gazebo" default="true" /><!-- Using Gazebo : Enables using HDT Robotic Arm in Gazebo Simulation with control-->
	<arg name="use_pp" default="true" /><!-- Using Path Planning : Enables using MoveIT & RViZ Path Planning -->
	<arg name="connection" default="tcpcoms" /><!-- Using Path Planning : Enables using MoveIT & RViZ Path Planning -->
	 <!-- "loopback" - is for computer use , "tcpcoms" - is for real arm use  -->


	<!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
	<include file="$(find hdt_rlb_moveit)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>
	
	<!-- load urdf, start state publishers -->
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
		<param name="use_gui" value="$(arg load_gui)" />
		<rosparam param="/source_list">[/hdt_adroit_coms/joint_cmd]</rosparam>
	</node>

	<!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
	<include file="$(find hdt_rlb_moveit)/launch/move_group.launch">
		<arg name="allow_trajectory_execution" value="false"/>  
		<arg name="fake_execution" value="false"/>
	</include>

	<!-- launch joystick -->
	<node name="xbox_joystick" pkg="joy" type="joy_node">
		<param name="dev" value="/dev/input/js0" />
		<param name="deadzone" value="0.15" />
	</node>

	<!-- launch hocu -->
	<node name="hocu_driver" pkg="hdt_hocu_driver" type="hocu_driver">
		<param name="port" value="8766" />
	</node>

	<!-- launch coms node -->
	<include file="$(find hdt_adroit_driver)/launch/hdt_adroit_coms.launch">
		<arg name="coms_interface" value="$(arg connection)" /> 
		<arg name="ip_address" value="192.168.1.45" />
		<arg name="port" value="8765" />
		<arg name="impedance_enabled" value="false" />
		<arg name="console_enabled" value="false" />
	</include>
	
	<!-- launch rviz -->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find hdt_rlb_description)/launch/hdt_rlb.rviz"/>

	<!-- launch axis camera -->
	<group if="$(arg use_axis)">
		<node name="axis_camera" pkg="axis_camera" type="axis.py" >
			<param name="hostname" value="192.168.1.47" /> 
			<param name="password" value="root" />
		</node>
	</group>

	<!-- launch leap motion --> 
	<group if="$(arg use_leap)">
		<node name="leap_motion" pkg="leap_motion" type="sender.py" /> 
	</group>

	<!-- launch arm control node -->
	<node name="arm_control" pkg="hdt_arm_control"  type="arm_control" respawn="false" output="screen">
		<param name="use_hocu" value="$(arg use_hocu)" />
		<param name="use_gazebo" value="$(arg use_gazebo)" />
		<param name="use_gui" value="$(arg use_gui)" /> 
		<param name="use_pp" value="$(arg use_pp)" /> 
		<param name="use_leap" value="$(arg use_leap)" /> 
	</node>
</launch>

